// @dxc-technology/halstack-react/file-input/FileItem.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _useTheme = _interopRequireDefault(require("../useTheme"));

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6, _templateObject7, _templateObject8, _templateObject9, _templateObject10;

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var deleteIcon = /*#__PURE__*/_react["default"].createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  width: "24",
  height: "24",
  viewBox: "0 0 24 24",
  fill: "currentColor"
}, /*#__PURE__*/_react["default"].createElement("path", {
  d: "M0 0h24v24H0V0z",
  fill: "none"
}), /*#__PURE__*/_react["default"].createElement("path", {
  d: "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
}));

var errorIcon = /*#__PURE__*/_react["default"].createElement("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  height: "24px",
  viewBox: "0 0 24 24",
  width: "24px",
  fill: "currentColor"
}, /*#__PURE__*/_react["default"].createElement("path", {
  d: "M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
}));

var FileItem = function FileItem(_ref) {
  var mode = _ref.mode,
      multiple = _ref.multiple,
      _ref$name = _ref.name,
      name = _ref$name === void 0 ? "" : _ref$name,
      _ref$error = _ref.error,
      error = _ref$error === void 0 ? "" : _ref$error,
      showPreview = _ref.showPreview,
      preview = _ref.preview,
      type = _ref.type,
      numFiles = _ref.numFiles,
      onDelete = _ref.onDelete,
      tabIndex = _ref.tabIndex;
  var colorsTheme = (0, _useTheme["default"])();
  var isImage = type.includes("image");

  var getIconAriaLabel = function getIconAriaLabel() {
    if (type.includes("video")) {
      return "video";
    }

    if (type.includes("audio")) {
      return "audio";
    }

    return "file";
  };

  return /*#__PURE__*/_react["default"].createElement(_styledComponents.ThemeProvider, {
    theme: colorsTheme.fileInput
  }, /*#__PURE__*/_react["default"].createElement(Container, {
    mode: mode,
    multiple: multiple,
    error: error,
    showPreview: showPreview,
    numFiles: numFiles
  }, showPreview && (isImage ? /*#__PURE__*/_react["default"].createElement(ImagePreview, {
    src: preview,
    alt: name
  }) : /*#__PURE__*/_react["default"].createElement(IconPreviewContainer, {
    error: error,
    "aria-label": getIconAriaLabel()
  }, /*#__PURE__*/_react["default"].createElement(IconPreview, {
    error: error
  }, preview))), /*#__PURE__*/_react["default"].createElement(FileItemContent, null, /*#__PURE__*/_react["default"].createElement(FileItemContainer, null, /*#__PURE__*/_react["default"].createElement(FileName, {
    mode: mode,
    multiple: multiple,
    error: error,
    showPreview: showPreview,
    numFiles: numFiles
  }, name), error && /*#__PURE__*/_react["default"].createElement(ErrorIcon, {
    "aria-label": "Error"
  }, errorIcon), /*#__PURE__*/_react["default"].createElement(DeleteIcon, {
    error: error,
    onClick: function onClick() {
      return onDelete(name);
    },
    "aria-label": "Remove ".concat(name),
    tabIndex: tabIndex
  }, deleteIcon)), error && (multiple || numFiles > 1) && /*#__PURE__*/_react["default"].createElement(ErrorMessage, null, error))));
};

var Container = _styledComponents["default"].div(_templateObject || (_templateObject = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  flex-direction: row;\n  padding: ", ";\n  background-color: ", ";\n  border-radius: ", ";\n  width: ", ";\n  min-height: ", ";\n  border-color: ", ";\n  border-width: ", ";\n  border-style: ", ";\n  display: flex;\n  justify-content: center;\n"])), function (props) {
  return props.showPreview ? "calc(8px - ".concat(props.theme.fileItemBorderThickness, ") 8px") : "calc(8px - ".concat(props.theme.fileItemBorderThickness, ") 8px calc(8px - ").concat(props.theme.fileItemBorderThickness, ") 16px");
}, function (props) {
  return props.error && props.theme.errorFileItemBackgroundColor;
}, function (props) {
  return props.theme.fileItemBorderRadius;
}, function (props) {
  return props.mode === "file" && !props.multiple && props.numFiles === 1 ? "calc(230px - 26px)" : !props.showPreview ? "calc(320px - 26px)" : props.showPreview && "calc(320px - 18px)";
}, function (props) {
  return props.mode === "file" && !props.multiple && props.numFiles === 1 || !props.showPreview && !props.error ? "calc(40px - 18px)" : !props.showPreview && props.error ? "calc(59px - 18px)" : "calc(64px - 18px)";
}, function (props) {
  return props.error ? props.theme.errorFileItemBorderColor : props.theme.fileItemBorderColor;
}, function (props) {
  return props.theme.fileItemBorderThickness;
}, function (props) {
  return props.theme.fileItemBorderStyle;
});

var FileItemContent = _styledComponents["default"].div(_templateObject2 || (_templateObject2 = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  flex-direction: column;\n"])));

var FileItemContainer = _styledComponents["default"].div(_templateObject3 || (_templateObject3 = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  flex-direction: row;\n  align-items: center;\n"])));

var ImagePreview = _styledComponents["default"].img(_templateObject4 || (_templateObject4 = (0, _taggedTemplateLiteral2["default"])(["\n  width: 48px;\n  height: 48px;\n  object-fit: contain;\n  margin-right: 12px;\n  border-radius: 2px;\n"])));

var IconPreviewContainer = _styledComponents["default"].div(_templateObject5 || (_templateObject5 = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  margin-right: 12px;\n  background-color: ", ";\n  width: 48px;\n  height: 48px;\n  border-radius: 2px;\n  color: ", ";\n"])), function (props) {
  return props.error ? props.theme.errorFilePreviewBackgroundColor : props.theme.filePreviewBackgroundColor;
}, function (props) {
  return props.error ? props.theme.errorFilePreviewIconColor : props.theme.filePreviewIconColor;
});

var IconPreview = _styledComponents["default"].div(_templateObject6 || (_templateObject6 = (0, _taggedTemplateLiteral2["default"])([""])));

var FileName = _styledComponents["default"].span(_templateObject7 || (_templateObject7 = (0, _taggedTemplateLiteral2["default"])(["\n  color: ", ";\n  width: ", ";\n  text-overflow: ellipsis;\n  overflow: hidden;\n  white-space: nowrap;\n  font-family: ", ";\n  font-size: ", ";\n  font-weight: ", ";\n  line-height: ", ";\n"])), function (props) {
  return props.theme.fileNameFontColor;
}, function (props) {
  return props.mode === "file" && !props.multiple && props.error && props.numFiles === 1 ? "calc(230px - 76px)" : props.mode === "file" && !props.multiple && !props.error && props.numFiles === 1 ? "calc(230px - 50px)" : !props.showPreview && !props.error ? "calc(320px - 52px)" : !props.showPreview && props.error ? "calc(320px - 76px)" : props.showPreview && props.error ? "calc(320px - 128px)" : props.showPreview && !props.error && "calc(320px - 102px)";
}, function (props) {
  return props.theme.fileItemFontFamily;
}, function (props) {
  return props.theme.fileItemFontSize;
}, function (props) {
  return props.theme.fileItemFontWeight;
}, function (props) {
  return props.theme.fileItemLineHeight;
});

var ErrorIcon = _styledComponents["default"].span(_templateObject8 || (_templateObject8 = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  flex-wrap: wrap;\n  align-content: center;\n  padding: 3px;\n  height: 18px;\n  width: 18px;\n  color: #d0011b;\n"])));

var DeleteIcon = _styledComponents["default"].button(_templateObject9 || (_templateObject9 = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  flex-wrap: wrap;\n  align-content: center;\n  height: 24px;\n  width: 24px;\n  font-size: 1rem;\n  font-family: ", ";\n  border: 1px solid transparent;\n  border-radius: 2px;\n  margin-left: 4px;\n  background-color: transparent;\n  box-shadow: 0 0 0 2px transparent;\n  padding: 3px;\n  cursor: pointer;\n  color: ", ";\n  svg {\n    line-height: 18px;\n  }\n  &:hover {\n    background-color: ", ";\n  }\n  &:focus {\n    outline: none;\n    box-shadow: 0 0 0 2px ", ";\n  }\n  &:focus-visible {\n    outline: none;\n    box-shadow: 0 0 0 2px ", ";\n  }\n  &:active {\n    background-color: ", ";\n  }\n"])), function (props) {
  return props.theme.fontFamily;
}, function (props) {
  return props.theme.deleteFileItemColor;
}, function (props) {
  return props.theme.hoverDeleteFileItemBackgroundColor;
}, function (props) {
  return props.theme.focusDeleteFileItemBorderColor;
}, function (props) {
  return props.theme.focusDeleteFileItemBorderColor;
}, function (props) {
  return props.theme.activeDeleteFileItemBackgroundColor;
});

var ErrorMessage = _styledComponents["default"].span(_templateObject10 || (_templateObject10 = (0, _taggedTemplateLiteral2["default"])(["\n  color: ", ";\n  font-family: ", ";\n  font-size: ", ";\n  font-weight: ", ";\n  line-height: ", ";\n"])), function (props) {
  return props.theme.errorMessageFontColor;
}, function (props) {
  return props.theme.errorMessageFontFamily;
}, function (props) {
  return props.theme.errorMessageFontSize;
}, function (props) {
  return props.theme.errorMessageFontWeight;
}, function (props) {
  return props.theme.errorMessageLineHeight;
});

var _default = FileItem;
exports["default"] = _default;
