// @dxc-technology/halstack-react/main.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BackgroundColorProvider", {
  enumerable: true,
  get: function get() {
    return _BackgroundColorContext.BackgroundColorProvider;
  }
});
Object.defineProperty(exports, "DxcAccordion", {
  enumerable: true,
  get: function get() {
    return _Accordion["default"];
  }
});
Object.defineProperty(exports, "DxcAccordionGroup", {
  enumerable: true,
  get: function get() {
    return _AccordionGroup["default"];
  }
});
Object.defineProperty(exports, "DxcAlert", {
  enumerable: true,
  get: function get() {
    return _Alert["default"];
  }
});
Object.defineProperty(exports, "DxcApplicationLayout", {
  enumerable: true,
  get: function get() {
    return _ApplicationLayout["default"];
  }
});
Object.defineProperty(exports, "DxcBadge", {
  enumerable: true,
  get: function get() {
    return _Badge["default"];
  }
});
Object.defineProperty(exports, "DxcBleed", {
  enumerable: true,
  get: function get() {
    return _Bleed["default"];
  }
});
Object.defineProperty(exports, "DxcBox", {
  enumerable: true,
  get: function get() {
    return _Box["default"];
  }
});
Object.defineProperty(exports, "DxcBulletedList", {
  enumerable: true,
  get: function get() {
    return _BulletedList["default"];
  }
});
Object.defineProperty(exports, "DxcButton", {
  enumerable: true,
  get: function get() {
    return _Button["default"];
  }
});
Object.defineProperty(exports, "DxcCard", {
  enumerable: true,
  get: function get() {
    return _Card["default"];
  }
});
Object.defineProperty(exports, "DxcCheckbox", {
  enumerable: true,
  get: function get() {
    return _Checkbox["default"];
  }
});
Object.defineProperty(exports, "DxcChip", {
  enumerable: true,
  get: function get() {
    return _Chip["default"];
  }
});
Object.defineProperty(exports, "DxcDateInput", {
  enumerable: true,
  get: function get() {
    return _DateInput["default"];
  }
});
Object.defineProperty(exports, "DxcDialog", {
  enumerable: true,
  get: function get() {
    return _Dialog["default"];
  }
});
Object.defineProperty(exports, "DxcDropdown", {
  enumerable: true,
  get: function get() {
    return _Dropdown["default"];
  }
});
Object.defineProperty(exports, "DxcFileInput", {
  enumerable: true,
  get: function get() {
    return _FileInput["default"];
  }
});
Object.defineProperty(exports, "DxcFlex", {
  enumerable: true,
  get: function get() {
    return _Flex["default"];
  }
});
Object.defineProperty(exports, "DxcHeading", {
  enumerable: true,
  get: function get() {
    return _Heading["default"];
  }
});
Object.defineProperty(exports, "DxcInset", {
  enumerable: true,
  get: function get() {
    return _Inset["default"];
  }
});
Object.defineProperty(exports, "DxcLink", {
  enumerable: true,
  get: function get() {
    return _Link["default"];
  }
});
Object.defineProperty(exports, "DxcNavTabs", {
  enumerable: true,
  get: function get() {
    return _NavTabs["default"];
  }
});
Object.defineProperty(exports, "DxcNumberInput", {
  enumerable: true,
  get: function get() {
    return _NumberInput["default"];
  }
});
Object.defineProperty(exports, "DxcPaginator", {
  enumerable: true,
  get: function get() {
    return _Paginator["default"];
  }
});
Object.defineProperty(exports, "DxcParagraph", {
  enumerable: true,
  get: function get() {
    return _Paragraph["default"];
  }
});
Object.defineProperty(exports, "DxcPasswordInput", {
  enumerable: true,
  get: function get() {
    return _PasswordInput["default"];
  }
});
Object.defineProperty(exports, "DxcProgressBar", {
  enumerable: true,
  get: function get() {
    return _ProgressBar["default"];
  }
});
Object.defineProperty(exports, "DxcQuickNav", {
  enumerable: true,
  get: function get() {
    return _QuickNav["default"];
  }
});
Object.defineProperty(exports, "DxcRadioGroup", {
  enumerable: true,
  get: function get() {
    return _RadioGroup["default"];
  }
});
Object.defineProperty(exports, "DxcResultsetTable", {
  enumerable: true,
  get: function get() {
    return _ResultsetTable["default"];
  }
});
Object.defineProperty(exports, "DxcSelect", {
  enumerable: true,
  get: function get() {
    return _Select["default"];
  }
});
Object.defineProperty(exports, "DxcSlider", {
  enumerable: true,
  get: function get() {
    return _Slider["default"];
  }
});
Object.defineProperty(exports, "DxcSpinner", {
  enumerable: true,
  get: function get() {
    return _Spinner["default"];
  }
});
Object.defineProperty(exports, "DxcSwitch", {
  enumerable: true,
  get: function get() {
    return _Switch["default"];
  }
});
Object.defineProperty(exports, "DxcTable", {
  enumerable: true,
  get: function get() {
    return _Table["default"];
  }
});
Object.defineProperty(exports, "DxcTabs", {
  enumerable: true,
  get: function get() {
    return _Tabs["default"];
  }
});
Object.defineProperty(exports, "DxcTag", {
  enumerable: true,
  get: function get() {
    return _Tag["default"];
  }
});
Object.defineProperty(exports, "DxcTextInput", {
  enumerable: true,
  get: function get() {
    return _TextInput["default"];
  }
});
Object.defineProperty(exports, "DxcTextarea", {
  enumerable: true,
  get: function get() {
    return _Textarea["default"];
  }
});
Object.defineProperty(exports, "DxcToggleGroup", {
  enumerable: true,
  get: function get() {
    return _ToggleGroup["default"];
  }
});
Object.defineProperty(exports, "DxcTypography", {
  enumerable: true,
  get: function get() {
    return _Typography["default"];
  }
});
Object.defineProperty(exports, "DxcWizard", {
  enumerable: true,
  get: function get() {
    return _Wizard["default"];
  }
});
Object.defineProperty(exports, "HalstackContext", {
  enumerable: true,
  get: function get() {
    return _HalstackContext["default"];
  }
});
Object.defineProperty(exports, "HalstackLanguageContext", {
  enumerable: true,
  get: function get() {
    return _HalstackContext.HalstackLanguageContext;
  }
});
Object.defineProperty(exports, "HalstackProvider", {
  enumerable: true,
  get: function get() {
    return _HalstackContext.HalstackProvider;
  }
});

var _Alert = _interopRequireDefault(require("./alert/Alert"));

var _Accordion = _interopRequireDefault(require("./accordion/Accordion"));

var _Button = _interopRequireDefault(require("./button/Button"));

var _Card = _interopRequireDefault(require("./card/Card"));

var _Checkbox = _interopRequireDefault(require("./checkbox/Checkbox"));

var _Dialog = _interopRequireDefault(require("./dialog/Dialog"));

var _Dropdown = _interopRequireDefault(require("./dropdown/Dropdown"));

var _Slider = _interopRequireDefault(require("./slider/Slider"));

var _Switch = _interopRequireDefault(require("./switch/Switch"));

var _Tabs = _interopRequireDefault(require("./tabs/Tabs"));

var _ProgressBar = _interopRequireDefault(require("./progress-bar/ProgressBar"));

var _Spinner = _interopRequireDefault(require("./spinner/Spinner"));

var _Table = _interopRequireDefault(require("./table/Table"));

var _Box = _interopRequireDefault(require("./box/Box"));

var _Tag = _interopRequireDefault(require("./tag/Tag"));

var _Paginator = _interopRequireDefault(require("./paginator/Paginator"));

var _Wizard = _interopRequireDefault(require("./wizard/Wizard"));

var _Link = _interopRequireDefault(require("./link/Link"));

var _Heading = _interopRequireDefault(require("./heading/Heading"));

var _ResultsetTable = _interopRequireDefault(require("./resultsetTable/ResultsetTable"));

var _Chip = _interopRequireDefault(require("./chip/Chip"));

var _ApplicationLayout = _interopRequireDefault(require("./layout/ApplicationLayout"));

var _ToggleGroup = _interopRequireDefault(require("./toggle-group/ToggleGroup"));

var _AccordionGroup = _interopRequireDefault(require("./accordion-group/AccordionGroup"));

var _Badge = _interopRequireDefault(require("./badge/Badge"));

var _TextInput = _interopRequireDefault(require("./text-input/TextInput"));

var _PasswordInput = _interopRequireDefault(require("./password-input/PasswordInput"));

var _DateInput = _interopRequireDefault(require("./date-input/DateInput"));

var _NumberInput = _interopRequireDefault(require("./number-input/NumberInput"));

var _Textarea = _interopRequireDefault(require("./textarea/Textarea"));

var _Select = _interopRequireDefault(require("./select/Select"));

var _FileInput = _interopRequireDefault(require("./file-input/FileInput"));

var _RadioGroup = _interopRequireDefault(require("./radio-group/RadioGroup"));

var _Bleed = _interopRequireDefault(require("./bleed/Bleed"));

var _Inset = _interopRequireDefault(require("./inset/Inset"));

var _QuickNav = _interopRequireDefault(require("./quick-nav/QuickNav"));

var _NavTabs = _interopRequireDefault(require("./tabs-nav/NavTabs"));

var _Flex = _interopRequireDefault(require("./flex/Flex"));

var _Typography = _interopRequireDefault(require("./typography/Typography"));

var _Paragraph = _interopRequireDefault(require("./paragraph/Paragraph"));

var _BulletedList = _interopRequireDefault(require("./bulleted-list/BulletedList"));

var _HalstackContext = _interopRequireWildcard(require("./HalstackContext"));

var _BackgroundColorContext = require("./BackgroundColorContext");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
